test/test_testing.py

```diff
@@ -4112,185 +4104,223 @@ def test_send_signal(self):
 
-    def test_push_and_pull_non_utf8_data(self):
+    def test_push_and_pull_non_utf8_data(
+        self,
+        pebble_dir: str,
+        client: typing.Union[_TestingPebbleClient, pebble.Client],
+    ):
         self._test_push_and_pull_data(
+            pebble_dir,
+            client,
             original_data='日本語',  # "Japanese" in Japanese
             encoding='sjis',
             stream_class=io.StringIO)
 
-    def _test_push_and_pull_data(self,
-                                 original_data: typing.Union[str, bytes],
-                                 encoding: typing.Optional[str],
-                                 stream_class: typing.Union[typing.Type[io.BytesIO],
-                                                            typing.Type[io.StringIO]]):
-        client = self.client
+    def _test_push_and_pull_data(
+        self,
+        pebble_dir: str,
+        client: typing.Union[_TestingPebbleClient, pebble.Client],
+        original_data: typing.Union[str, bytes],
+        encoding: typing.Optional[str],
+        stream_class: typing.Union[typing.Type[io.BytesIO], typing.Type[io.StringIO]],
+    ):
         # We separate out the calls to make it clearer to type checkers what's happening.
         if encoding is None:
-            client.push(f"{self.prefix}/test", original_data)
+            client.push(f"{pebble_dir}/test", original_data)
         else:
-            client.push(f"{self.prefix}/test", original_data, encoding=encoding)
-        with client.pull(f"{self.prefix}/test", encoding=encoding) as infile:
+            client.push(f"{pebble_dir}/test", original_data, encoding=encoding)
+        with client.pull(f"{pebble_dir}/test", encoding=encoding) as infile:
             received_data = infile.read()
         assert original_data == received_data
 
         # We also support file-like objects as input, so let's test that case as well.
         if encoding is None:
             stream_class = typing.cast(typing.Type[io.BytesIO], stream_class)
             small_file = stream_class(typing.cast(bytes, original_data))
-            client.push(f"{self.prefix}/test", small_file)
+            client.push(f"{pebble_dir}/test", small_file)
         else:
             stream_class = typing.cast(typing.Type[io.StringIO], stream_class)
             small_file = stream_class(typing.cast(str, original_data))
-            client.push(f"{self.prefix}/test", small_file, encoding=encoding)
-        with client.pull(f"{self.prefix}/test", encoding=encoding) as infile:
+            client.push(f"{pebble_dir}/test", small_file, encoding=encoding)
+        with client.pull(f"{pebble_dir}/test", encoding=encoding) as infile:
             received_data = infile.read()
         assert original_data == received_data
 
-    def test_push_bytes_ignore_encoding(self):
+    def test_push_bytes_ignore_encoding(
+        self,
+        pebble_dir: str,
+        client: typing.Union[_TestingPebbleClient, pebble.Client],
```

Comment:
1. Yes, because in real pebble test the client is real, not a mocked client. See the other already closed PR or `test_real_pebble.py`.
2. Will do, thanks!